plan/refiner.go

```diff
@@ -437,11 +444,65 @@ func (c *conditionChecker) checkScalarFunction(scalar *expression.ScalarFunction
 	case ast.IsNull, ast.IsTruth, ast.IsFalsity:
 		return c.checkColumn(scalar.Args[0])
 	case ast.UnaryNot:
+		// Don't support "not like" and "not in" convert to access conditions.
+		if s, ok := scalar.Args[0].(*expression.ScalarFunction); ok {
```

Comment:
What if `s` is OR expression, e.g. `select * from x where not (a like "a" or a like "ab")`?